### PR TITLE
Fix: double quotes not being handled properly in interactive mode

### PR DESCRIPTION
**Pull request by attfarhan**
Fixes https://github.com/sourcegraph/sourcegraph/issues/8418.

~Now, we will always wrap ~the content and message~ filter inputs in double quotes, and escape any double quotes within the input. As was the case before, we will show the quoted and escaped values when the filter input is non-editable, but we will show the unescaped and unquoted value when it is editable.~

This means that for all filters, even those that we previously _required_ users to include quotes (e.g. `after:`, `before:`), users should _never_ include quotes to delimit their input, otherwise the quotes will be included in the match and result in an error or unexpected match.

### CODE REVIEW

File: web/src/search/input/helpers.ts [hunk 1]
```diff
@@ -53,3 +53,9 @@ export function convertPlainTextToInteractiveQuery(
 
     return { filtersInQuery: newFiltersInQuery, navbarQuery: newNavbarQuery }
 }
+
+// Escapes non-escaped double quotes in a string.
+// The regex specifies to match quotes which are not preceded by a backslash.
+// Not all browser support negative lookbehinds, so the regexp works by
+// including all characters, and \" in a negated group.
+export const escapeDoubleQuotes = (input: string): string => input.replace(/([^(?:(?!\\"))\w\W]\s\S)/g, '\\"')
```
Comment (lguychard) [inline web/src/search/input/helpers.ts]: I think we want to escape _all_ double quotes, not just _unescaped_ double quotes (eg. simply always JSON.stringify).

Consider the following valid pattern, used to search for an escaped double quote: [`\\"`](https://sourcegraph.com/search?q=content:%5C%22&patternType=literal). In order to yield the expected results, it needs to be stringified to `"\\\""`, not `"\""`.

Comment (rvantonder) [reply web/src/search/input/helpers.ts]: Yes, it shouldn't be necessary to hand roll this. Escaping a string should be possible using some language builtin or library.

**Comment by rvantonder** [comment]
> This means that for all filters, even those that we previously required users to include quotes (e.g. after:, before:), users should never include quotes to delimit their input

Yes, agree with this decision. I would push that we should just quote/escape all filters (not just those for content), and you will be guaranteed to not run into any trouble when converting back and forth to raw queries. All filters can take quoted values, and will unquote in the backend.

**Comment by lguychard** [comment]
> I would push that we should just quote/escape all filters (not just those for content), and you will be guaranteed to not run into any trouble when converting back and forth to raw queries.

+1 👍 

> All filters can take quoted values, and will unquote in the backend.

Huh, TIL that indeed, `archived:"only"` works just fine 😅

**Pull request merged**